client/src/components/ratingsReviews/RatingBreakdown.jsx

```diff
@@ -2,66 +2,64 @@ import React from 'react';
 import RatingProgress from './RatingProgress.jsx';
 import StarRating from '../shared/StarRating.jsx';
 
-
 const RatingBreakdown = (props) => {
-  const { reviews, ratings, recommended } = props;
+  const { ratings, recommended, handleRatingProgressFilter } = props;
   const stars = [1, 1, 1, .8, 0];
   let key = 0;
 
-  const oneBar = parseInt(ratings[1]) || 0;
-  const oneStar = oneBar * 1 || 0;
-  const twoBar = parseInt(ratings[2]) || 0;
-  const twoStar = twoBar * 2 || 0;
-  const threeBar = parseInt(ratings[3]) || 0;
-  const threeStar = threeBar * 3 || 0;
-  const fourBar = parseInt(ratings[4]) || 0;
-  const fourStar = fourBar * 4 || 0;
-  const fiveBar = parseInt(ratings[5]) || 0;
-  const fiveStar = fiveBar * 5 || 0;
 
-  const sum = oneStar + twoStar + threeStar + fourStar + fiveStar;
-  const sumBar = oneBar + twoBar + threeBar + fourBar + fiveBar;
-  const averageRating = sum / sumBar;
+  if (ratings !== null && ratings) {
+    const oneBar = parseInt(ratings[1]) || 0;
+    const oneStar = oneBar * 1 || 0;
+    const twoBar = parseInt(ratings[2]) || 0;
```

Comment:
This whole section is a mystery to me. Whatever you're doing, I'm sure there's a way to calculate `star` values from `bar` values, or vice versa, without hard-coding a name for each value. For all I know, this could be a simple calculation in `render`. On each couple of lines, you're saying the same number five times, e.g. "3": `threeBar` (1), `parseInt(ratings[3]` (1), `threeStar` (3) `threeBar` again (4), `* 3` (5). Refactoring this would obviously be a pain, and that's kind of the problem/point: hard-coded things are hard to refactor.

Reply:
I started with hardcoded code in order to have placeholders for all the filters my widget has. I've refactored part of it and will work on your suggestion from PR# 56